src/perforceManager.ts

```diff
@@ -272,6 +272,21 @@ class PerforceManager {
         // @ts-ignore
         return _.map(noneDeletedFiles, file => file.depotFile);
     }
+
+    private async getLabelOrChangelistWithWorkspace(fullPath: string, workspace: IPerforceWorkspace): Promise<string> {
+        const cstat = await this.p4.cmd(`cstat "${fullPath}"@${workspace.client} `);
+        if (cstat.error) {
+            logger.error("Failed to get changelist for file", {fullPath, cstat});
+            return null;
+        }
+        const changelist = cstat?.stat?.[0]?.change;
+        const labels = await this.p4.cmd(`labels "${fullPath}"@${changelist}`);
+        if (labels.error) {
+            logger.error("Failed to get labels for file", {fullPath, changelist, labels});
+        }
```

Comment:
Shouldn't you check whether changelist has a valid valid before running the labels cmd?

Reply:
If it is invalid we will get errors but I can check anyway

Reply:
Meaning there will be cstat.error